pysmt/solvers/solver.py

```diff
@@ -59,18 +61,8 @@ def is_sat(self, formula):
         assert formula in self.environment.formula_manager, \
                "Formula does not belong to the current Formula Manager"
 
-        use_solving_under_assumption = False
```

Comment:
Why can't we rely on solving under assumptions instead of breaking the contract of `is_sat`? 

If a solver does not support incrementality nor solving under assumptions, then an error can be thrown...